.github/workflows/release.yml

```diff
@@ -17,58 +20,33 @@ on:
     # branches:
     # - ag/release
     tags:
-    - '[0-9]+.[0-9]+.[0-9]+'
+      - "[0-9]+.[0-9]+.[0-9]+"
 jobs:
   create-release:
     name: create-release
     runs-on: ubuntu-latest
-    # env:
-      # Set to force version number, e.g., when no tag exists.
-      # RG_VERSION: TEST-0.0.0
+    outputs:
+      upload_url: ${{ steps.release.outputs.upload_url }}
     steps:
-      - name: Create artifacts directory
-        run: mkdir artifacts
-
-      - name: Get the release version from the tag
-        if: env.RG_VERSION == ''
-        run: |
-          # Apparently, this is the right way to get a tag name. Really?
-          #
-          # See: https://github.community/t5/GitHub-Actions/How-to-get-just-the-tag-name/m-p/32167/highlight/true#M1027
-          echo "::set-env name=RG_VERSION::${GITHUB_REF#refs/tags/}"
-          echo "version is: ${{ env.RG_VERSION }}"
-
       - name: Create GitHub release
         id: release
         uses: actions/create-release@v1
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          tag_name: ${{ env.RG_VERSION }}
-          release_name: ${{ env.RG_VERSION }}
-
-      - name: Save release upload URL to artifact
-        run: echo "${{ steps.release.outputs.upload_url }}" > artifacts/release-upload-url
-
-      - name: Save version number to artifact
-        run: echo "${{ env.RG_VERSION }}" > artifacts/release-version
-
-      - name: Upload artifacts
-        uses: actions/upload-artifact@v1
-        with:
-          name: artifacts
-          path: artifacts
+          tag_name: ${{ github.ref }}
+          release_name: ${{ github.ref }}
```

Comment:
Can you say why this works?

Oh, I see, you moved the logic down below.

Reply:
It's right from the examples of [`@actions/create-release`](https://www.github.com/actions/create-release) and [`@actions/upload-release-asset`](https://www.github.com/actions/upload-release-asset). They're smart enough to strip the unnecessary part and leave only tag name. Cannot find the resource where it was mentioned, but it works as we can see :man_shrugging: